src/lib/comms/sol-lwm2m-server.c

```diff
@@ -1295,26 +1295,41 @@ send_management_packet(struct sol_lwm2m_server *server,
     return -ENOMEM;
 }
 
-/**
- * This checks if the path has the following form: /2/0/1
- */
-static bool
-is_resource_set(const char *path)
+enum lwm2m_path_props {
+    PATH_IS_INVALID = (1 << 0),
+    PATH_HAS_OBJECT =  (1 << 1),
+    PATH_HAS_INSTANCE = (1 << 2),
+    PATH_HAS_RESOURCE = (1 << 3)
+};
```

Comment:
Just out of curiosity: is there any advantage in using left-shifts instead of `(1)`, `(2)`, `(4)` and `(8)`?